### DISCUSSION TITLE
Multi-Scale Simulation Using Moose HeatTransfer and Phase Field Modules

### DISCUSSION THREAD

**Discussion by BingkangLi** [idaholab/moose#27339] [Q&A Modules: General] [answered]
Hi,
I'm exploring the implementation of a multi-scale simulation using Moose's HeatTransfer and Phase Field modules. The workflow entails computing the temperature field within a parent application. Subsequently, the temperature data from each unit will be transmitted to a child application. This child application will then utilize the received temperature data to execute a phase field simulation. Following the phase field simulation, the effective thermal conductivity will be computed based on the results obtained.

In this setup, each unit will run a phase field simulation independently. To manage this process, I plan to employ CentroidMultiApp to seamlessly integrate the child and parent applications.

However, the phase field represents a dynamic process where the computation of each phase field iteration relies on the results obtained from the previous step. Therefore, my concern pertains to whether CentroidMultiApp can ensure that the subsequent invocation of the phase field can accurately access the results from the previous iteration.

I welcome any insights or recommendations on how to ensure the smooth integration and continuity of results within this multi-scale simulation framework.

**Comment by GiudGiud** [comment]
Hello

>  from the previous step

you mean time step right? Or do you mean fixed point iteration between the heat transfer and phase field simulations?

If you use a TransientMultiApp for the phase field  simulations, they will naturally restart from where they were on the previous step.
If you need these solves to be transients, you can specify a Transient executioner with no time derivative term, that is the same as a steady solve.

Guillaume

**Reply by BingkangLi** [reply]
Hi @GiudGiud  I mean each element in the heat conduction FEM simulation will run a phase-field simulation. I examined the source code of `CentroidMultiApp` and found that it is public from the `TransientMultiApp`. If I use `CentroidMultiApp`, can I somehow automatically achieve the 'naturally restart from where they were on the previous step' behavior on each element of the heat conduction FEM simulation? 

the `CentroidMultiApp.h` file：

> class CentroidMultiApp : public TransientMultiApp, public BlockRestrictable
{
public:
  static InputParameters validParams();
  CentroidMultiApp(const InputParameters & parameters);
protected:
  virtual void fillPositions() override;
};

Binglang Li

**Reply by GiudGiud** [reply, marked as answer]
this is correct. Yes CentroidMultiApp will have the same behavior as TransientMultiApp in that regard.

**Reply by GiudGiud** [reply]
note that if a multiapp at every element centroid is "too many" multiapps, you can use the 'positions' arguments of the transient multiapp instead

**Reply by BingkangLi** [reply]
I tested the `CentroidMultiApp` ,you are right, thank you very much. :-)
